server/src/jobs/impl/nut/process.cpp

```diff
@@ -25,54 +25,94 @@ Process::~Process()
     }
 }
 
-Expected<void> Process::init(const std::string& address, uint16_t port)
+Expected<void> Process::setupSnmp(const std::string& address, uint16_t port)
 {
-    std::string driver;
-    if (m_protocol == "nut_snmp") {
-        driver = "snmp-ups";
-    }
-    if (m_protocol == "nut_xml_pdc") {
-        driver = "netxml-ups";
+    if (!port) {
+        port = 161;
     }
 
-    if (driver.empty()) {
-        return unexpected("Protocol {} is not supported", m_protocol);
+    std::string toconnect = fmt::format("{}:{}", address, port);
+
+    if (auto path = findExecutable("snmp-ups")) {
+        // clang-format off
+        m_process = std::unique_ptr<fty::Process>(new fty::Process(*path, {
+            "-s", "discover",
+            "-x", fmt::format("port={}", toconnect),
+            "-d", "1"
```

Comment:
d1 may not be enough to ensure that all SNMP data are gathered!
I've always advocated for 2 at least ;)